### PR TITLE
fix: .d.ts definitions for core and app

### PR DESCRIPTION
**Pull request by yurist38**
Hello! It's a partial fix for #2926 including TypeScript type definitions for `netlify-cms-core` and `netlify-cms-app`. 

**Summary**
I'm using Netlify CMS in TS project, so I needed types for it. I first wrote them in my project, so these types are partially tested in my project. However, there is still room for improvement since some types are defined as `any` which is not good. If I'll get more details about those types I'll open a separate PR with the fix. Also, please take a look and maybe you can tell me straight away what kind of values we can expect there, so I could change it right away before merging this PR.
Also, I've updated `package.json` files for these two packages so the types will be automatically added to the next NPM release.

**Test plan**
Well, who is testing before going to production?! Let's have some fun of unpredictability :)
Joking aside, It should work as expected. I did tests in my project with updates of the `node_modules` folder.

**A picture of a cute animal (not mandatory but encouraged)**
Why not... Sharing with you the pic of my lovely cat (her usual posture).
IMG_20191110_125433
(it's me on Friday night 😄)

### CODE REVIEW

File: packages/netlify-cms-core/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    preview_context?: string;
+    auth_type?: string;
+    app_id?: string;
+    api_root?: string;
+    base_url?: string;
+    auth_endpoint?: string;
+  }
+
+  export interface CmsConfig {
+    backend?: CmsBackendType | CmsBackendParam;
```
Comment (erezrokah) [inline packages/netlify-cms-core/index.d.ts]: According to the schema here https://github.com/netlify/netlify-cms/blob/09ca1a7f0391b19b6d6065fcde93fc5a855a0f60/packages/netlify-cms-core/src/constants/configSchema.js#L32

backend should be an object

**Comment by erezrokah** [comment]
This is going to be very helpful for dev experience. If you need a reference for the types you can use the config schema I mentioned in my comment

File: packages/netlify-cms-core/index.d.ts (hunk 1 above)
Comment (yurist38) [reply packages/netlify-cms-core/index.d.ts]: Good remark, thanks! Didn't see the schema, to be honest... 😄  Let me fix this.

**Comment by yurist38** [comment]
@erezrokah I've refactored typings to match the schema, thanks for sharing it!

**Pull request merged**